### PR TITLE
[SPARK-25955][TEST] Porting JSON tests for CSV functions

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

In the PR, I propose to port existing JSON tests from `JsonFunctionsSuite` that are applicable for CSV, and put them to `CsvFunctionsSuite`. In particular:
- roundtrip `from_csv` to `to_csv`, and `to_csv` to `from_csv`
- using `schema_of_csv` in `from_csv`
- Java API `from_csv`
- using `from_csv` and `to_csv` in exprs.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

**Review by dongjoon-hyun — COMMENTED** [review]
@MaxGekk . Sorry, but `Porting` seems to be not the best way to do this.
Could you refactor this by introducing new test helper functions?
I believe you can test `from_json/csv/avro` with the same functions.

File: sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala [hunk 1]
```diff
@@ -86,4 +86,82 @@ class CsvFunctionsSuite extends QueryTest with SharedSQLContext {
 
     checkAnswer(df.select(to_csv($"a", options)), Row("26/08/2015 18:00") :: Nil)
   }
+
+  test("from_csv uses DDL strings for defining a schema - java") {
+    val df = Seq("""1,"haa"""").toDS()
+    checkAnswer(
+      df.select(
+        from_csv($"value", lit("a INT, b STRING"), new java.util.HashMap[String, String]())),
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala]: The only difference is `from_csv` and `from_json`.

File: sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala [hunk 2]
```diff
@@ -86,4 +86,82 @@ class CsvFunctionsSuite extends QueryTest with SharedSQLContext {
...
+      "parsed",
+      StructType(Seq(
+        StructField("_c0", DoubleType, true),
+        StructField("_c1", IntegerType, true),
+        StructField("_c2", StringType, true))))))
+
+    assert(out.schema == expected)
+  }
+
+  test("Support to_csv in SQL") {
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala]: This is only for double check that the functions are available/(and work) from expressions in Scala. Probably we can make the test smaller.

**Comment by dongjoon-hyun** [comment]
Yes. It would be great if we do that in this PR.

When I did the similar thing for ORC (`port tests from Parquet to ORC`, `port from old ORC to new ORC`). I received the same comments.

**Comment by dongjoon-hyun** [comment]
Ur, maybe, I'm not clear to the point. The refactoring scope of this PR is limited to the new tests here.
```
test("from_csv uses DDL strings for defining a schema - java")
test("roundtrip to_csv -> from_csv")
test("roundtrip from_csv -> to_csv")
test("infers schemas of a CSV string and pass to to from_csv")
test("Support to_csv in SQL")
test("Support from_csv in SQL")
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**